drivers/flash/flash_stm32h7x.c

```diff
@@ -33,11 +35,27 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #error Flash driver on M4 core is not supported yet
 #endif
 
-#if defined(CONFIG_MULTITHREADING)
+#define REAL_FLASH_SIZE   KB(LL_GetFlashSize())
```

Comment:
I would propose `REAL_FLASH_SIZE_KB`, to know it is in kB.

Reply:
Done